kratos/includes/gid_io.h

```diff
@@ -61,12 +61,28 @@ enum MultiFileFlag {SingleFile, MultipleFiles};
 
 class KRATOS_API(KRATOS_CORE) GidIOBase : public IO {
 
-    protected:
+protected:
     /**
      * Counter of live GidIO instances
      * (to ensure GiD_PostInit and GiD_PostDone are properly called)
      */
-    static int msLiveInstances;
+    int data;
+
+    // Private constructor so that no objects can be created.
+    GidIOBase() {
+        data = 0;
+    }
+
+public:
+    static GidIOBase& GetInstance();
+
+    int GetData();
+    void SetData(int data);
+
+private:
+    static void Create();
+
+    static GidIOBase* mpInstance;
```

Comment:
Please add some Doxygen documentation to these methods.

Reply:
To be honest I don't know what they do, they were already there. I supose they control the number of uses? Not sure...

Reply:
I will add it to the singleton ones. Those I know for sure